server/szurubooru/func/posts.py

```diff
@@ -485,14 +478,61 @@ def test_sound(post: model.Post, content: bytes) -> None:
             update_post_flags(post, flags)
 
 
+def purge_post_signature(post: model.Post) -> None:
+    old_signature = (
+        db.session
+        .query(model.PostSignature)
+        .filter(model.PostSignature.post_id == post.post_id)
+        .one_or_none())
+    if old_signature:
+        db.session.delete(old_signature)
```

Comment:
Maybe it could be shortened to `db.session.filter(…).delete()`?

Reply:
Seems to work.